### ISSUE TITLE
linux distribution compatibility and CI packaging

### ISSUE THREAD

**Issue by tcecyk** [openandroidinstaller-dev/openandroidinstaller#41]
just an early heads up, not wanting to divert you from the important work that you started - you can close this or refer to when the packaging topic comes up again. See at the end I think this will resolve itself for most.

I gave the v.0.21-alpha a run in debian-sid that is more current than the ubuntu in the github workflow (20.04 currently). At start it complains about missing rendering libraries, but the error is misleading, it's about library compatibility.
```
libEGL warning: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
```
pointing it to swrast specifically with `LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/ LIBGL_DRIVERS_PATH=/usr/lib/x86_64-linux-gnu/dri/ LIBGL_ALWAYS_SOFTWARE=1 ./openandroidinstaller` gives it away
```
libEGL warning: MESA-LOADER: failed to open swrast: /tmp/_MEIuW7CIn/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /usr/lib/x86_64-linux-gnu/dri//swrast_dri.so) (search paths /usr/lib/x86_64-linux-gnu/dri/, suffix _dri)
```
I see the installer is extracting static libraries into /tmp/_M<random>. The `libstdc++.so.6` therin uses the host mesa libraries that are complied against more current libc than what you'd have from the build by the github workflow. Because they use symbols the shipped libstdc++ hasn't those can't be loaded. There is an issue on it at flutter that I commented - https://github.com/flutter/flutter/issues/76178#issuecomment-1327519899

One remedy can be to use the `ubuntu-22.04` tag [in the CI](https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/1202bb1e5cdc31fb6b6ade652d033022dc74a63b/.github/workflows/manual-build.yml#L13) instead of `ubuntu-latest` that points to 20.04 still. But I see the tag will roll over to 22.04 any day now: https://github.com/actions/runner-images/issues/6399

With time this can creep back in when users move ahead in their versions. Packaging for flatpak / snap base-packages of those libraries and/or for distribution versions once you move towards a v1.0 can help.

**Comment by tsterbak** [comment]
Hi @tcecyk, thanks a lot for reporting the issue!

I'm not very experienced with packaging cross-platform desktop applications, but I will look at the issue and hopefully figure it out then :grinning: 
Once the tool is getting closer to v1.0, the plan would be to package it as flatpak either way. If you have any(more) suggestions or tips on packaging I would greatly appreciate it :slightly_smiling_face: 

Lets hope it resolves itself :grin: 

A bit unrelated to that, but did you also try to run the OpenAndroidInstaller from python? And maybe even tried to install some custom ROM with it?

**Comment by tcecyk** [comment]
> .. did you also try to run the OpenAndroidInstaller .. ?

sure! I'm surveying installers I can contribute to, I like it and will give feedback soon.

I opened this issue for others hitting it and checking the issues / searching (after finally reading about symbol versioning myself haha). If you'll come back to packaging later separately I close this?

**Comment by tsterbak** [comment]
That sounds great! I would be happy to have you contributing! Let me know if I can help you in any way.

I think you can leave it open, then I will remember when I come back to packaging.

**Comment by tcecyk** [comment]
I see `ubuntu-latest` rolled over to 22.04 mid december and you're building both, the (now) old 20.04 target and latest 22.04, great!

**Comment by tsterbak** [comment]
Hey, thank you! Yes, I'm building for both now, of course the 22.04. build doesn't work on 20.04. So I'm a bit concerned about the build approach...

Did you give the installer a try in the meantime? :)

**Comment by sudwhiwdh** [comment]
> I'm not very experienced with packaging cross-platform desktop applications, but I will look at the issue and hopefully figure it out then.
Once the tool is getting closer to v1.0, the plan would be to package it as flatpak either way. If you have any(more) suggestions or tips on packaging I would greatly appreciate it.

@barthalion @bilelmoussaoui

There is already a branch started.
https://github.com/openandroidinstaller-dev/openandroidinstaller/tree/flatpak
Could you imagine to support this project with your knowledge to make it accessible as a Flatpak?

**Comment by tsterbak** [comment]
This issue will be resolved by the flatpak release #144
